### PR TITLE
Migrate from requests/responses to httpx/respx

### PR DESCRIPTION
**Pull request by janwijbrand**
This includes many stylistic changes that you may or may not like. Consider it an excercise on my behalf getting to know the codebase and ideas. Please let me know what you think.

### CODE REVIEW

**Comment by janwijbrand** [comment]
Note, I'm far from "done" with the stylistic but also structural changes. But let's talk first :-)

**Comment by janwijbrand** [comment]
I've used 3.8+ syntax... Travis is failing because of it.

File: orthanc_ext/event_dispatcher.py [hunk 1]
```diff
@@ -39,7 +41,7 @@ def OnChange(change_type, resource_type, resource_id):
         return_values = []
         for handler in handlers:
             event = ChangeEvent(change_type, resource_type, resource_id)
-            return_values.append(handler(event, requests_session))
+            return_values.append(handler(event, client))
```
Comment (janwijbrand) [inline orthanc_ext/event_dispatcher.py:44]: `session`? `client`?

File: orthanc_ext/requests_utilities.py [hunk 2]
```diff
@@ -11,8 +11,8 @@ def get_certificate(config):
     return config.get('SslCertificate', False)
 
 
-def create_internal_requests_session(base_url, token, cert=False):
-    session = sessions.BaseUrlSession(base_url)
-    session.headers['Authorization'] = token
-    session.verify = cert
-    return session
+def create_internal_requests_session(base_url, token='', cert=None) -> httpx.Client:
```
Comment (janwijbrand) [inline orthanc_ext/requests_utilities.py:14]: do not use name of impl detail

File: tests/test_auto_forward.py [hunk 3]
```diff
@@ -1,69 +1,67 @@
...
 orthanc = OrthancApiHandler()
-session = event_dispatcher.create_internal_requests_session('https://localhost:8042', '')
+session = create_internal_requests_session('https://localhost:8042')
 
 
-@responses.activate
-def test_autoforward_on_match_shall_start_start_modality_store(caplog):
-    responses.add(responses.POST, 'https://localhost:8042/modalities/pacs/store', body='study-uuid', status=200)
+def register_and_trigger_handler(matchers):
+    register_event_handlers({orthanc.ChangeType.STABLE_STUDY: forward_dicom(matchers)},
```
Comment (janwijbrand) [inline tests/test_auto_forward.py:16]: is the `({` because of the yapf coalesce brackets setting?

File: docs/usage.rst [hunk 4]
```diff
@@ -14,13 +14,13 @@ specified in orthanc.json or as an environment variable for the Osimis docker im
        logging.warn(evt.resource_id)
 
     event_dispatcher.register_event_handlers({orthanc.ChangeType.STABLE_STUDY: log_event}, orthanc_module=orthanc,
-                                             requests_session=event_dispatcher.create_session(orthanc))
+                                             session=event_dispatcher.create_session(orthanc))
```
Comment (walcovanloon) [inline docs/usage.rst:17]: we settled on client (?)

File: orthanc_ext/event_dispatcher.py [hunk 5]
```diff
@@ -39,14 +41,14 @@ def OnChange(change_type, resource_type, resource_id):
...
```
Comment (walcovanloon) [inline orthanc_ext/event_dispatcher.py:44]: client here and below, but ^^^ we use session

**Pull request merged**